app/models/parser/twitter_profile.rb

```diff
@@ -11,8 +11,8 @@ def patterns
         [
           /^https?:\/\/(www\.)?twitter\.com\/(?<username>[\w\d]+)(\?+.*)$/,
           /^https?:\/\/(0|m|mobile)\.twitter\.com\/(?<username>[\w\d]+)(\?+.*)$/,
-          /^https?:\/\/(www\.)?twitter\.com\/(?<username>[^\/]+)$/,
-          /^https?:\/\/(0|m|mobile)\.twitter\.com\/(?<username>[^\/]+)$/
+          /^https?:\/\/(www\.)?twitter\.com\/(?<username>[\w\d]{4,15})$/,
+          /^https?:\/\/(0|m|mobile)\.twitter\.com\/(?<username>[\w\d]{4,15})$/
```

Comment:
Request: The two first regular expressions can still match usernames with less than 4 characters or more than 15 characters, so we need to fix them.

Tip: `\w` already includes digits, so `\w\d` can be replaced by just `\w`.

Suggestion: Combine all four regular expressions into a single one... something like this (untested):
```
/^https?:\/\/((0|m|mobile|www)\.)?twitter\.com\/(?<username>[\w]{4,15})(\?.*)?$/
```

Question: What about trailing slashes? Are they stripped by normalization or some previous step?

Reply:
> Question: What about trailing slashes? Are they stripped by normalization or some previous step?

They were not, I updated the regex to also take trailing slashes into account